presto-product-tests/src/main/java/io/prestosql/tests/kafka/TestKafkaWritesSmokeTest.java

```diff
@@ -0,0 +1,272 @@
+                        "('kasia', 9223372036854775807, 2147483647, 32767, null, null, null)",
+                KAFKA_CATALOG,
+                SCHEMA_NAME,
+                ALL_DATATYPES_CSV_TABLE_NAME
+        ))).updatedRowsCountIsEqualTo(5);
+
+        assertThat(query(format("select * from %s.%s.%s", KAFKA_CATALOG, SCHEMA_NAME, ALL_DATATYPES_CSV_TABLE_NAME))).containsOnly(
+                row("jasio", 9223372036854775807L, 2147483647, 32767, 127, 1234567890.123456789, true),
+                row("stasio", -9223372036854775808L, -2147483648, -32768, -128, -1234567890.123456789, false),
+                row(null, null, null, null, null, null, null),
+                row("krzysio", 9223372036854775807L, 2147483647, 32767, 127, 1234567890.123456789, false),
+                row("kasia", 9223372036854775807L, 2147483647, 32767, null, null, null));
+    }
+
+    private static final String ALL_DATATYPES_JSON_TABLE_NAME = "write_all_datatypes_json";
+    private static final String ALL_DATATYPES_JSON_TOPIC_NAME = "write_all_datatypes_json";
+
+    private static class AllDataTypesJsonTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    SCHEMA_NAME + "." + ALL_DATATYPES_JSON_TABLE_NAME,
+                    ALL_DATATYPES_JSON_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(AllDataTypesJsonTable.class)
+    public void testInsertJsonTable()
+    {
+        assertThat(query(format(
+                "insert into %s.%s.%s values " +
+                        "('ala ma kota', " +
+                        " 9223372036854775807, " +
+                        " 2147483647, " +
+                        " 32767, " +
+                        " 127, " +
+                        " 1234567890.123456789, " +
+                        " true, " +
+                        " TIMESTAMP '2018-02-09 13:15:16', " +
+                        " TIMESTAMP '2018-02-09 13:15:17', " +
+                        " TIMESTAMP '2018-02-09 13:15:18', " +
+                        " TIMESTAMP '2018-02-09 13:15:19', " +
+                        " TIMESTAMP '2018-02-09 13:15:20', " +
+                        " DATE '2018-02-11', " +
+                        " DATE '2018-02-13', " +
+                        " TIME '13:15:16', " +
+                        " TIME '13:15:17', " +
+                        " TIME '13:15:18', " +
+                        " TIME '13:15:20', " +
+                        // different because product test framework converts to java.sql.Timestamp
+                        " TIMESTAMP '2018-02-09 13:15:18 UTC', " +
```

Comment:
Which column is it?

if we write from PT and read from PT, why would there be a difference?

Also, not sure what "different" is refering. This is INSERT statetement, so the "different" comment would be more appropriate on the SELECT query below

Reply:
Sorry, I misplaced the comment. It's meant to apply to the SELECT statement. The query assertions use a java.sql.Timestamp and we need to pass a LocalDateTime to it and hence the LocalDateTime value looks different from the expected value.

Reply:
> The query assertions use a java.sql.Timestamp and we need to pass a LocalDateTime to it and hence the LocalDateTime value looks different from the expected value.

This is unlikely the reason. With some great dose of carefulness, the two can be used "together" well.
If the value looks different, something is likely wrong. Or we stripped zone in some mapping, whereupon a comment should indicate that.

Reply:
Ok, let me go through the code once with some debug logging to confirm.